app/controllers/concerns/account_concern.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module AccountConcern
+  extend ActiveSupport::Concern
+
+  ACCOUNT_SESSION_HEADER_NAME = "HTTP_GOVUK_ACCOUNT_SESSION"
+  ACCOUNT_END_SESSION_HEADER_NAME = "HTTP_GOVUK_ACCOUNT_END_SESSION"
+  ACCOUNT_SESSION_DEV_COOKIE_NAME = "govuk_account_session"
+
+  included do
+    before_action :fetch_account_session_header
+    before_action :set_account_session_header
+    helper_method :logged_in?
+
+    attr_accessor :account_session_header
+  end
+
+  def logged_in?
+    account_session_header.present?
+  end
+
+  def fetch_account_session_header
+    @account_session_header =
+      if request.headers.to_h[ACCOUNT_SESSION_HEADER_NAME]
+        request.headers.to_h[ACCOUNT_SESSION_HEADER_NAME]
+      elsif Rails.env.development?
+        cookies[ACCOUNT_SESSION_DEV_COOKIE_NAME]
+      end
+  end
+
+  def set_account_session_header(govuk_account_session = nil)
+    @account_session_header = govuk_account_session if govuk_account_session
+    response.headers[ACCOUNT_SESSION_HEADER_NAME] = @account_session_header
```

Comment:
I think it's just the request headers that use the `HTTP_GOVUK_ACCOUNT_SESSION` format, and the response headers use the `GOVUK-Account-Session` format.  See [this bit of govuk_ab_testing](https://github.com/alphagov/govuk_ab_testing/blob/master/lib/govuk_ab_testing/ab_test.rb#L29-L36).